### DISCUSSION TITLE
Number Badge

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#214] [Component suggestions]
<img width="480" alt="Visueel voorbeeld van het number badge component" src="https://github.com/nl-design-system/backlog/assets/545207/e0e96768-2404-499e-ac22-bd6e72bcddf3">

We willen graag meer leren over het **Number Badge** component.

- In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?
- Welke combinaties of binnen welk component wordt de Number badge gebruikt?
- Tot hoe ver tel je?
- We zijn ook benieuwd of/hoe het visueel verborgen label wordt geschreven.

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1706610945457019)

---

## Naam
Number Badge

## Doel
Toont een klein label dat de aandacht trekt en informeert over een aantal.

## Gerelateerde componenten
[Data Badge](https://github.com/orgs/nl-design-system/discussions/307), [Dot Badge](https://github.com/orgs/nl-design-system/discussions/308), [Status Badge](https://github.com/orgs/nl-design-system/discussions/290)

## Richtlijnen
Acceptatiecriteria: [NL Design System Acceptatiecriteria voor Number Badge](https://nldesignsystem.nl/number-badge/)

## Onderzoek
[Adding an Item to a Shopping Cart - Nielsen Norman Group](https://www.nngroup.com/articles/cart-feedback/)
Artikel over het geven van feedback wanneer een item aan een winkelmand is toegevoegd.

[Notification badge - HMRC design resources](https://design.tax.service.gov.uk/hmrc-design-patterns/notification-badge/)
Onder 'Research' wordt geschreven dat gebruikers begrijpen wat het is en dat het niet voor afleiding zorgt.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-feedback-badge--docs)
- [GitHub](https://github.com/Amsterdam/design-system/blob/develop/packages/css/src/components/badge/badge.scss)
- [Figma](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=10946-16973&mode=design&t=K7p7iPfIMqbmNOt1-4)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-05-30 om 11 45 21](https://github.com/nl-design-system/backlog/assets/545207/a66e82aa-5e26-4f2c-876f-340e49447dd1)

**Figma**
![Scherm­afbeelding 2024-01-30 om 11 51 00](https://github.com/nl-design-system/backlog/assets/545207/95e75383-ca0a-4059-9aa1-9087b20450a8)

Ook gebruikt binnen een 'Filter button'.
![Scherm­afbeelding 2024-01-30 om 11 51 08](https://github.com/nl-design-system/backlog/assets/545207/7ac7e0f1-1b9c-41b0-bbbf-1d9836c3e2d3)

---

### Notities
- Wordt gezien als het Badge component met de mogelijkheid om hier enkel nummers in te tonen. Hierdoor veel opties voor kleur.

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-numberbadge--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/NumberBadge)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=1540-4477&mode=design&t=E3sXUxfTcCDOB8Hu-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=1233-4272&mode=design&t=e7lUoBUNlKeXkLTc-4)

---

### Screenshots / Varianten
![Scherm­afbeelding 2024-01-30 om 12 04 26](https://github.com/nl-design-system/backlog/assets/545207/0a6fd197-1ca3-4d01-93d5-75b114a078da)

Ook gebruikt in combinatie met menu items
![Scherm­afbeelding 2024-01-30 om 12 07 08](https://github.com/nl-design-system/backlog/assets/545207/54e9ac00-6afe-4bb5-8ee4-47695569fccc)

---

### Notities
- Wordt gecombineerd met Dot badge
- Paar mooie Do's and Dont's

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

- [Documentatie website](https://dso-toolkit.nl/62.8.4/components/signals/badge)
- [Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/core-badge--plain&viewMode=story)
- [Github](https://github.com/dso-toolkit/dso-toolkit/tree/master/packages/dso-toolkit/src/components/badge)

---

### Screenshots / Varianten

**Plain**
![Scherm­afbeelding 2024-01-30 om 12 19 47](https://github.com/nl-design-system/backlog/assets/545207/c9eef5b6-8f8d-41b8-99bf-ffb670c39bff)

---

### Notities
- Wordt gezien als het Badge component met de mogelijkheid om als 'teller' te dienen.
- In documentatie wordt enkel de plain variant getoond.
- Storybook toont opties in kleur op basis van status. In CSS classes ook als zodanig benoemd.

**Comment by jeffreylauwers** [comment]
## DUO

- [Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/badge)

---

### Screenshots / Varianten
**Notificaties**
Gebruik een badge met icoon voor notificaties.
![Scherm­afbeelding 2024-01-30 om 14 56 43](https://github.com/nl-design-system/backlog/assets/545207/581f8ee8-83a4-4c70-902f-10d26a7080c7)

---

### Notities
- Wordt gezien als het Badge component met de mogelijkheid om hier enkel nummers in te tonen. Hierdoor veel opties voor kleur. 
- Opties in kleur op basis van kleur. In CSS classes ook als zodanig benoemd.
- Wordt als voorbeeld gecombineerd met een icoon waarbij de badge een rode kleur heeft.

**Comment by jeffreylauwers** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/css-utrecht-badge-counter--docs)
- [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=7927-10565&mode=design&t=o4sYDyhvvPbx4b1O-4)

---

### Screenshots / Varianten

**Light blue**
![Scherm­afbeelding 2024-01-30 om 15 20 32](https://github.com/nl-design-system/backlog/assets/545207/b465eff9-a73d-458b-9e2d-e6fe75c0d743)

**Blue**
![Scherm­afbeelding 2024-01-30 om 15 20 50](https://github.com/nl-design-system/backlog/assets/545207/d998d620-233e-41cf-96dd-33e9e7620b3e)

**Dark blue**
![Scherm­afbeelding 2024-01-30 om 15 20 54](https://github.com/nl-design-system/backlog/assets/545207/42f1be26-7c58-49b4-b8be-b71a6fcdb03d)

**Green blue**
![Scherm­afbeelding 2024-01-30 om 15 20 59](https://github.com/nl-design-system/backlog/assets/545207/380b46e9-0888-4265-9618-eea316fd2c1f)

**Red**
![Scherm­afbeelding 2024-01-30 om 15 21 04](https://github.com/nl-design-system/backlog/assets/545207/ec3bcd1c-1412-45a7-920b-351c5c0285ab)

**Dark yellow**
![Scherm­afbeelding 2024-01-30 om 15 21 08](https://github.com/nl-design-system/backlog/assets/545207/abcb3363-a756-4916-8c44-fb08c1a73695)

---

### Notities
- Storybook op basis van Utecht counter badge.
- Opties in kleur op basis van kleur. In CSS classes ook als zodanig benoemd.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-badge-counter--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/badge-counter)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=1233-4272&mode=design&t=DOSJkeqQP5dGs0uJ-4)

---

### Screenshots / Varianten
![Scherm­afbeelding 2024-01-30 om 15 32 50](https://github.com/nl-design-system/backlog/assets/545207/d366dee8-63e2-43ba-a582-e0a528774a12)

---

### Notities
- Gedrag beschreven bij meerdere cijfers en talen.

**Comment by Robbert** [comment]
Misschien moet het doel expliciet zeggen dat het gaat om een "aantal items van een specifiek iets"

Bijvoorbeeld: "Klein label dat informeert over een aantal items in een verzameling, waarvan de verzameling wordt beschreven in een gerelateerd component"

**Comment by Robbert** [comment]
Ik stel voor om de component "Number badge" te noemen, om meer in lijn te zijn met webstandaarden. Daar komt "counter" daar niet veel in voor.

- HTML:
  - "Number" komt voor in `type="number"`
- JavaScript:
  - `typeof count === "number"`
  - `new Number`
  - `Intl.NumberFormat`
- CSS:
  - `<number>` type in CSS
- SVG:
  - `SVGAnimatedNumber`

Counter komt met name voor in CSS, voor het nummeren van lijstjes: https://developer.mozilla.org/en-US/docs/Web/CSS/counter In XSLT heb je `<xsl:number>` die voor aantallen weergeven is, met het `count` attribuut specifiek voor lijsten en hoofdstuknummers.

Qua voorspelbaarheid van namen zou het ook mooi zijn om het in lijn te brengen met toekomstige componenten, zoals:

- [Number data](https://nl-design-system.github.io/utrecht/storybook-react/index.html?path=/docs/react-number-data--docs)
- [Number input](https://nldesignsystem.nl/number-input)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Number badge op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=number+badge)

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Number badge op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=number+badge)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Candidate ✨

Help je mee hem door de Hall of Fame stappen te krijgen?
[Number Badge op het Hall of Fame bord](https://github.com/orgs/nl-design-system/projects/30/views/1?filterQuery=number+badge)
